### PR TITLE
chore: remove redundant error codes

### LINKED ISSUE #1556: Unhelpful and unorganized error messages
We should organize all error messages and properly document them, so there is a consistent, helpful and descriptive set of error messages across the sdk instead of ones I experienced like "too small" with no value or details flagged.

> [!NOTE]
> - Incoming from https://github.com/FuelLabs/fuels-ts/discussions/1545

### PR DESCRIPTION
**Pull request by petertonysmith94**
Related to #1556 

---

- Removed redundant error codes.

### CODE REVIEW

**Review by Torres-ssf — APPROVED** [review]

File: packages/contract/src/contract-factory.ts [hunk 1]
```diff
@@ -92,8 +92,8 @@ export default class ContractFactory {
   createTransactionRequest(deployContractOptions?: DeployContractOptions) {
     const storageSlots = deployContractOptions?.storageSlots
       ?.map(({ key, value }) => ({
-        key: hexlifyWithPrefix(key, true),
-        value: hexlifyWithPrefix(value, true),
+        key: hexlifyWithPrefix(key),
+        value: hexlifyWithPrefix(value),
```
Comment (danielbate) [inline packages/contract/src/contract-factory.ts:96]: Added by mistake? Although that whole function is redundant, we can just use `hexlify`.

Comment (danielbate) [inline packages/contract/src/contract-factory.ts:96]: We could just call `hexlify` here?

File: packages/contract/src/util.ts [hunk 2]
```diff
@@ -64,17 +63,8 @@ export const getContractId = (
...
-    return hexlify(value);
-  }
-
-  if (isKnownHex) {
-    return hexlify(`0x${value}`);
-  }
-
-  throw new FuelError(FuelError.CODES.UNEXPECTED_HEX_VALUE, `Value should be hex string ${value}.`);
-};
+export const hexlifyWithPrefix = (value: string) =>
```
Comment (danielbate) [inline packages/contract/src/util.ts:69]: IMO this function is pretty redundant. Hexlifying a hex value will produce the same value. Defo an edge case and unlike but we risk this passing with `0xNotAHexVal`. May as well just call `hexlify`.

**Review by danielbate — CHANGES_REQUESTED** [review]
Just one thought about `hexlify` usage, rest looks good

File: packages/contract/src/contract-factory.ts (hunk 1 above)
Comment (petertonysmith94) [reply packages/contract/src/contract-factory.ts]: Took a lil time to think over this one and I do agree.

https://github.com/FuelLabs/fuels-ts/pull/2002/commits/e73dc54934d0e14b4172a4cfa60129d23258c407

**Review by danielbate — APPROVED** [review]

File: packages/contract/src/util.ts (hunk 2 above)
Comment (petertonysmith94) [reply packages/contract/src/util.ts:69]: Reverted in 23995888b8e82a863f03e60f4fd8896b7a2ab711.

As storage slots have key, values without the `0x` prefix, it [fails](https://github.com/FuelLabs/fuels-ts/actions/runs/8542109255/job/23403007707?pr=2002#step:3:1011).

**Review by danielbate — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

**Pull request merged**